src/main/java/com/coveros/selenified/element/check/Excludes.java

```diff
@@ -0,0 +1,200 @@
+     * as a mismatch. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support.
+     *
+     * @param expectedValue the expected value of the element
+     */
+    void text(String expectedValue);
+
+    default String checkText(String expectedValue, double waitFor, double timeTook) {
+        // record the action
+        getOutputFile().recordAction(getElement().prettyOutput() + EXCLUDES_TEXT + expectedValue + "</b>", waitFor);
+        // check for the object to the present on the page
+        String elementValue = getElement().get().text();
+        // record the result
+        if (elementValue == null || elementValue.contains(expectedValue)) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + HAS_TEXT + elementValue + "</b>", timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + HAS_TEXT + elementValue + "</b>", timeTook, Success.PASS);
+        }
+        return elementValue;
+    }
+
+    /**
+     * Verifies that the element's value does not contain the provided expected
+     * value. If the element isn't present or an input, this will constitute a
+     * failure, same as a mismatch. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     *
+     * @param expectedValue the expected value of the element
+     */
+    void value(String expectedValue);
+
+    default String checkValue(String expectedValue, double waitFor, double timeTook) {
+        // record the action and get our value
+        String elementValue = getValue(expectedValue, EXCLUDES_VALUE, waitFor);
+        if (elementValue == null || elementValue.contains(expectedValue)) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + HAS_VALUE + elementValue + "</b>", timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + HAS_VALUE + elementValue + "</b>", timeTook, Success.PASS);
+        }
+        return elementValue;
+    }
+
+    /**
+     * Verifies that the element's options do not contain the provided expected
+     * option. If the element isn't present or a select, this will constitute a
+     * failure, same as a mismatch. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     *
+     * @param option the option not expected in the list
+     */
+    void selectOption(String option);
+
+    default String[] checkSelectOption(String option, double waitFor, double timeTook) {
+        // record the action, and check for select
+        if (!isPresentSelect(getElement().prettyOutput() + " without the option <b>" + option +
+                "</b> available to be selected on the page", waitFor)) {
+            return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty array instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1168)